core/org.osate.aadl2.edit/src/org/osate/aadl2/provider/FlowImplementationItemProvider.java

```diff
@@ -136,9 +140,19 @@ protected EStructuralFeature getChildFeature(Object object, Object child) {
 	 * This returns FlowImplementation.gif.
 	 * <!-- begin-user-doc -->
 	 * <!-- end-user-doc -->
-	 * @generated
+	 * @generated NOT
 	 */
+	@Override
 	public Object getImage(Object object) {
+		if (((FlowImplementationImpl) object).getInEnd() == null
+				&& ((FlowImplementationImpl) object).getOutEnd() == null) {
+			return null;
+		} else if (((FlowImplementationImpl) object).getInEnd() == null) {
+			return overlayImage(object, getResourceLocator().getImage("full/obj16/FlowSource"));
+		} else if (((FlowImplementationImpl) object).getOutEnd() == null) {
+			return overlayImage(object, getResourceLocator().getImage("full/obj16/FlowSink"));
```

Comment:
Add icons for flow source/sink implementation (and I above the line like flow implementation).